### ISSUE TITLE
false positive for map_entry when two different maps used

### ISSUE THREAD

**Issue by madadam** [rust-lang/rust-clippy#4219]
Clippy wrongly triggers the map_entry lint in the following case (note the map used with `contains_key` is different from the one used with `insert`):

```rust
    let a: HashMap<String, String> = HashMap::new();
    let mut b: HashMap<String, String> = HashMap::new();

    let key = "foo".to_string();
    
    if !a.contains_key(&key) {
        b.insert(key, "baz".to_string());
    }
```

Link to the playground (click Tools -> Clippy): https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=b230b4300359f8ddaa4e2610d2e0dd1c

```
> cargo clippy -V
clippy 0.0.212 (265318d 2019-05-17)
```
